internal/pkg/service/stream/keboolasink/bridge/bridge_test.go

```diff
@@ -0,0 +1,214 @@
+	client := mocked.TestEtcdClient()
+	defRepo := d.DefinitionRepository()
+	storageRepo := d.StorageRepository()
+	fileRepo := storageRepo.File()
+	volumeRepo := storageRepo.Volume()
+
+	// Log etcd operations
+	var etcdLogs bytes.Buffer
+	rawClient := d.EtcdClient()
+	rawClient.KV = etcdlogger.KVLogWrapper(rawClient.KV, &etcdLogs, etcdlogger.WithMinimal())
+
+	// apiCtx - for operations triggered by an authorized API call
+	apiCtx := rollback.ContextWith(bgCtx, rollback.New(d.Logger()))
+	apiCtx = context.WithValue(apiCtx, dependencies.KeboolaProjectAPICtxKey, mocked.KeboolaProjectAPI())
+
+	// Register mocked responses
+	// -----------------------------------------------------------------------------------------------------------------
+	transport := mocked.MockedHTTPTransport()
+	{
+		bridgeTest.MockTokenStorageAPICalls(t, transport)
+		bridgeTest.MockBucketStorageAPICalls(t, transport)
+		bridgeTest.MockTableStorageAPICalls(t, transport)
+		bridgeTest.MockFileStorageAPICalls(t, clk, transport)
+	}
+
+	// Register active volumes
+	// -----------------------------------------------------------------------------------------------------------------
+	{
+		session, err := concurrency.NewSession(client)
+		require.NoError(t, err)
+		defer func() { require.NoError(t, session.Close()) }()
+		test.RegisterWriterVolumes(t, bgCtx, volumeRepo, session, 1)
+	}
+
+	// Create parent branch, source, sink, file, slice
+	// -----------------------------------------------------------------------------------------------------------------
+	{
+		branch := test.NewBranch(branchKey)
+		require.NoError(t, defRepo.Branch().Create(&branch, clk.Now(), by).Do(apiCtx).Err())
+		source := test.NewSource(sourceKey)
+		require.NoError(t, defRepo.Source().Create(&source, clk.Now(), by, "Create source").Do(apiCtx).Err())
+	}
+
+	// Create sink, file, slice
+	// -----------------------------------------------------------------------------------------------------------------
+	var createSinkEtcdLogs string
+	{
+		sink := test.NewKeboolaTableSink(sinkKey)
+		etcdLogs.Reset()
+		require.NoError(t, defRepo.Sink().Create(&sink, clk.Now(), by, "Create sink").Do(apiCtx).Err())
+		createSinkEtcdLogs = etcdLogs.String()
+	}
+	{
+		// Storage API calls
+		assert.Equal(t, 1, transport.GetCallCountInfo()["POST https://connection.keboola.local/v2/storage/tokens"])
+		assert.Equal(t, 1, transport.GetCallCountInfo()["POST https://connection.keboola.local/v2/storage/branch/456/buckets"])
+		assert.Equal(t, 1, transport.GetCallCountInfo()["POST https://connection.keboola.local/v2/storage/branch/456/buckets/in.c-bucket/tables-definition"])
+		assert.Equal(t, 1, transport.GetCallCountInfo()["POST https://connection.keboola.local/v2/storage/branch/456/files/prepare"])
+		transport.ZeroCallCounters()
+	}
+	{
+		// Logs
+		mocked.DebugLogger().AssertJSONMessages(t, `
+{"level":"info","message":"creating bucket","bucket.key":"456/in.c-bucket"}
```

Comment:
There is no message with `tokens` interaction. Do we assume that it is not much room to make mistake?

Reply:
:thinking: @Matovidlo there are already test for tokens logs:

https://github.com/keboola/keboola-as-code/blob/df9a9169aa303c0a1c715ceb3b839551533d7e60/internal/pkg/service/stream/keboolasink/bridge/bridge_test.go#L113-L114

https://github.com/keboola/keboola-as-code/blob/df9a9169aa303c0a1c715ceb3b839551533d7e60/internal/pkg/service/stream/keboolasink/bridge/bridge_test.go#L178-L179

https://github.com/keboola/keboola-as-code/blob/df9a9169aa303c0a1c715ceb3b839551533d7e60/internal/pkg/service/stream/keboolasink/bridge/bridge_test.go#L146-L147

....